src/printer.js

```diff
@@ -1110,7 +1110,16 @@ function genericPrintNoParens(path, options, print) {
             n.expression.type === "LogicalExpression");
 
       if (shouldInline) {
-        return concat(["{", path.call(print, "expression"), "}"]);
+        const shouldPutBracketOnNextLine =
+          n.expression.type === "LogicalExpression" ||
+          n.expression.type === "ConditionalExpression";
+
+        return concat([
+          "{",
+          path.call(print, "expression"),
+          shouldPutBracketOnNextLine ? softline : "",
```

Comment:
If you want `}` to be on the next line, it seems like that shouldn't just happen when there is indentation, but if *any* of the expression breaks (if it spans multiple lines). Why can't we just put `softline` here all the time?